executors/cpp/datetime_fmt.cpp

```diff
@@ -0,0 +1,257 @@
+    if (option_item) {
+      timeStyle_str = json_object_get_string(option_item);
+      time_style = stringToEStyle(timeStyle_str);
+      cout << "# timeStyle: " << timeStyle_str << " " << time_style << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "calendar");
+    if (option_item) {
+      dateStyle_str = json_object_get_string(option_item);
+      cout << "# calendar: " << calendar_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "timezone");
+    if (option_item) {
+      timezone_str = json_object_get_string(option_item);
+      cout << "# timezone: " << timezone_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "era");
+    if (option_item) {
+      era_str = json_object_get_string(option_item);
+      cout << "# era: " << era_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "year");
+    if (option_item) {
+      year_str = json_object_get_string(option_item);
+      cout << "# year: " << year_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "month");
+    if (option_item) {
+      month_str = json_object_get_string(option_item);
+      cout << "# month: " << month_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "weekday");
+    if (option_item) {
+      weekday_str = json_object_get_string(option_item);
+      cout << "# weekdat: " << weekday_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "day");
+    if (option_item) {
+      day_str = json_object_get_string(option_item);
+      cout << "# day: " << day_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "hour");
+    if (option_item) {
+      hour_str = json_object_get_string(option_item);
+      cout << "# hour: " << hour_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "minute");
+    if (option_item) {
+      minute_str = json_object_get_string(option_item);
+      cout << "# minute: " << minute_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "second");
+    if (option_item) {
+      second_str = json_object_get_string(option_item);
+      cout << "# second: " << second_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "fractional_seconds_digits");
+    if (option_item) {
+      fractional_seconds_digits = json_object_get_int(option_item);
+      cout << "# fractional seconds digits: " << fractional_seconds_digits << endl;
+    }
+
+  }
+
+  json_object *date_skeleton_obj = json_object_object_get(json_in, "datetime_skeleton");
+  if (date_skeleton_obj) {
+    // Data specifies a date time skeleton. Make a formatter based on this.
+    string skeleton_string = json_object_get_string(date_skeleton_obj);
+
+    cout << "# Skelecton = " << skeleton_string << endl;
+
+    UnicodeString u_skeleton = skeleton_string.c_str();
```

Comment:
Apparently, this isn't the way to convert C String -> UnicodeString. You should use [this UnicodeString constructor](https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/classicu_1_1UnicodeString.html#a613a04db62048c98e32112e5864716db), and only if the text is ASCII. If the `char *` text is not ASCII, then you need to do a codepage conversion first before converting to UnicodeString.

(I asked Shane b/c I didn't see any `char *` arg `operator=()` for UnicodeString. He learned the above info from Markus, so you can double check with him to make sure that's correct.)

Reply:
Thanks. Constructor is now in place.